apps/backend/src/deposits/pos-deposit.service.ts

```diff
@@ -19,55 +19,36 @@ export class PosDepositService {
   findAll(): Promise<POSDepositEntity[]> {
     return this.findAll();
   }
-  async getMerchantIdsByLocationId(event: ReconciliationEvent) {
-    const merchantIds = await Promise.all(
-      await this.posDepositRepo.manager.query(
-        `SELECT 
-        "Merchant ID" as merchant_id 
-      FROM 
-        master_location_data 
-      WHERE 
-        "GARMS Location"=${event?.location_id} 
-      AND "Type" 
-        !='Bank'`
-      )
-    );
-    return merchantIds?.map(({ merchant_id }: { merchant_id: string }) =>
-      parseInt(merchant_id)
-    );
-  }
 
-  async queryPOS(
-    event: ReconciliationEvent,
-    merchant_ids: number[]
+  async findPOSDeposits(
+    event: ReconciliationEvent
   ): Promise<POSDepositEntity[]> {
-    return await this.posDepositRepo.manager.query(`
-    SELECT 
-      pd.id, 
-      pd.transaction_amt, 
-      pd.match, 
-      pd.card_vendor, 
-      pd.transaction_date::varchar, 
-      pd.transaction_time::varchar, 
-      pm.method as method
-    FROM 
-      pos_deposit pd 
-    JOIN 
-      payment_method pm 
-    ON 
-      pm.method=pd.card_vendor 
-    WHERE 
-      transaction_date='${event?.date}'::date 
-    AND 
-      pd.program='${event?.program}'
-    AND 
-      pd.merchant_id IN (${merchant_ids}) 
-    AND 
-      pd.match=false 
-    ORDER BY 
-      pd.transaction_amt
-    `);
+    const merchant_ids =
+      await this.locationService.getMerchantIdsByLocationId(
+        event?.location_id || 0
+      ); // TODO: Fix || 0
+    return await this.posDepositRepo.find({
+      relationLoadStrategy: 'query',
+      relations: {
+        payment_method: true
+      },
+      where: {
+        transaction_date: event?.date,
+        metadata: {
+          program: event?.program
+        },
+        merchant_id: In(merchant_ids)
+      },
+      // Order by needs to be in this order for matching logic. 
+      // We need to batch them using order to ease matches
+      order: {
```

Comment:
this is ordering is necessary for EAGER match. so that the matches actually are working based on card vendor AND time